pax-lang/src/interpreter/mod.rs

```diff
@@ -0,0 +1,375 @@
+use computable::Computable;
+use pax_runtime_api::{pax_value::functions::call_function, PaxValue, Percent, Rotation, Size};
+use pax_runtime_api::{CoercionRules, Functions, Numeric};
+use pest::{
+    iterators::{Pair, Pairs},
+    pratt_parser::{self, PrattParser},
+};
+use property_resolution::IdentifierResolver;
+use std::{collections::HashMap, rc::Rc};
+
+use crate::{
+    deserializer::from_pax_ast, get_pax_pratt_parser, parse_pax_err, parse_pax_pairs, PaxParser,
+    Rule,
+};
+
+pub(crate) mod computable;
+pub mod property_resolution;
+mod tests;
+
+#[derive(PartialEq, Debug)]
+pub enum PaxExpression {
+    Primary(Box<PaxPrimary>),
+    Prefix(Box<PaxPrefix>),
+    Infix(Box<PaxInfix>),
+    Postfix(Box<PaxPostfix>),
+}
+
+#[derive(PartialEq, Debug)]
+pub enum PaxPrimary {
+    Literal(PaxValue),
+    Grouped(Box<PaxExpression>, Option<PaxUnit>),
```

Comment:
I think there is probably some reconciling we can do in the grammar in general. Tried to do minimal changes there since it snowballs into a lot of the parsing logic and didn't want it to block. 
 
I did think about doing that when I created that GroupedVariant but after considering saw a bit of complexity around it.  Specifically, if we made units a post-fix operator and kept literal_number_with_units, then pratt parsing would be ambiguous since {10px} would look like both 10px as a primary and 10 with px as a postfix. I didn't see a huge benefit from making the change that would warrant solving that ambiguity problem. Lmk if you see a benefit other than nice-ness of shape.